src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.CompilationManager.cs

```diff
@@ -61,10 +61,12 @@ private void ClearCompilationsWithAnalyzersCache()
             // we basically eagarly clear the cache on some known changes
             // to let CompilationWithAnalyzer go.
 
-            // we create new conditional weak table every time, it turns out 
-            // only way to clear ConditionalWeakTable is re-creating it.
-            // also, conditional weak table has a leak - https://github.com/dotnet/coreclr/issues/665
+            // we create new conditional weak table every time netstandard as that's the only way it has to clear it.
+#if NETSTANDARD
             _projectCompilationsWithAnalyzers = new ConditionalWeakTable<Project, CompilationWithAnalyzers?>();
```

Comment:
assigning like this means the old value gets GC'ed which runs the finalizer for the CWT.  the finalizer means everything this points at gets elevated to the next generation.